### PR TITLE
Fix cumulative transactions

### PR DESCRIPTION
**Pull request by aprotopopov**
## Overview

- add argument `div_coef` for `summary_data_from_transaction_data` to get exact values like for CDNOW summary (#100 could be closed)
- rewrite function `expected_cumulative_transactions` to fix bugs and improve performance
- add tests for `utils.expected_cumulative_transactions` with comparison to [R BTYD walktrough](https://cran.r-project.org/web/packages/BTYD/vignettes/BTYD-walkthrough.pdf)
- rewrite `test_plotting` with fixtures, although it slowed down tests a bit..

Travis CI failed due to plots, probably we need to recreate them.
I could rebase commits to get more clear structure if needed.

### CODE REVIEW

File: lifetimes/utils.py [hunk 1]
```diff
@@ -433,53 +440,74 @@ def expected_cumulative_transactions(model, transactions, datetime_col, customer
...
-        a string that represents the timestamp format. Useful if Pandas can't understand
-        the provided format.
+        a string that represents the timestamp format. Useful if Pandas can't
+        understand the provided format.
     freq: string, optional
         Default 'D' for days, 'W' for weeks, 'M' for months... etc. Full list here:
         http://pandas.pydata.org/pandas-docs/stable/timeseries.html#dateoffset-objects
     set_index_date: bool, optional
         when True set date as Pandas DataFrame index, default False - number of time units
+    div_coef: float, optional
```
Comment (CamDavidsonPilon) [inline lifetimes/utils.py]: this is a better docstring than the one above - want to copy this up there too?

Comment (CamDavidsonPilon) [reply lifetimes/utils.py]: if `div_coef` is a float, an expression like `i % div_coef == 0` is ambiguous about it's output, no?

File: lifetimes/utils.py [hunk 2]
```diff
@@ -197,9 +199,12 @@ def summary_data_from_transaction_data(transactions, customer_id_col, datetime_c
     datetime_format: string, optional
         a string that represents the timestamp format. Useful if Pandas can't understand
         the provided format.
-    freq: string
+    freq: string, optional
         Default 'D' for days, 'W' for weeks, 'M' for months... etc. Full list here:
         http://pandas.pydata.org/pandas-docs/stable/timeseries.html#dateoffset-objects
+    div_coef: float, optional
```
Comment (CamDavidsonPilon) [inline lifetimes/utils.py]: Can you think of a better name for `div_coef`? Something more representative?

File: tests/test_plotting.py [hunk 3]
```diff
@@ -13,22 +13,56 @@
...
-cd_data = load_cdnow_summary()
-bgf.fit(cd_data['frequency'], cd_data['recency'], cd_data['T'], iterative_fitting=1)
+
+@pytest.fixture()
+def cd_data():
+    return load_cdnow_summary()
+
+
+@pytest.fixture()
+def bgf(cd_data):
```
Comment (CamDavidsonPilon) [inline tests/test_plotting.py]: 👍 great call

File: tests/test_utils.py [hunk 4]
```diff
@@ -409,3 +446,70 @@ def test_expected_cumulative_transactions_dedups_inside_a_time_period(fitted_bg,
     by_week = utils.expected_cumulative_transactions(fitted_bg, example_transaction_data, 'date', 'id', 10, freq='W')
     by_day = utils.expected_cumulative_transactions(fitted_bg, example_transaction_data, 'date', 'id', 10, freq='D')
     assert (by_week['actual'] >= by_day['actual']).all()
+
+
+def test_expected_cumulative_transactions_equals_r_btyd_walktrough(df_cum_transactions):
+    # data from BTYD walkthrough
```
Comment (CamDavidsonPilon) [inline tests/test_utils.py]: can you all put in this doc string the script used to repro this data?

**Comment by CamDavidsonPilon** [comment]
> Travis CI failed due to plots, probably we need to recreate them.

👍 want me to update now, and then you can rebase from master?

**Comment by CamDavidsonPilon** [comment]
Very cool PR, thanks @aprotopopov. I left some comments. Most of it is around the nomenclature of the new param and documentation surrounding it.

File: lifetimes/utils.py (hunk 1 above)
Comment (aprotopopov) [reply lifetimes/utils.py]: Good catch it should be int with current logic.

File: lifetimes/utils.py (hunk 2 above)
Comment (aprotopopov) [reply lifetimes/utils.py]: What about `freq_multiplier`? Or maybe you could suggest a better one?

**Comment by aprotopopov** [comment]
@CamDavidsonPilon thanks for your comments. 

> want me to update now, and then you can rebase from master?
yes, let's update plots to have clear CI for the master. I'll try to rebase this branch then.

changed `div_coef` to `freq_multiplier`. Still don't know about more explicit name.

File: tests/test_utils.py [hunk 5]
```diff
@@ -77,6 +79,42 @@ def large_transaction_level_data_with_monetary_value():
...
+@pytest.fixture()
+def cdnow_transactions():
+    transactions = load_dataset('CDNOW_sample.txt', header=None, sep=r'\s+')
+    transactions.columns = ['id_total', 'id_sample', 'date', 'num_cd_purc',
+                            'total_value']
+    return transactions[['id_sample', 'date']]
+
+
+@pytest.fixture()
+def df_cum_transactions(cdnow_transactions):
```
Comment (CamDavidsonPilon) [inline tests/test_utils.py]: this is a great set of tests & test-fixtures to check your implementation 👏

**Comment by CamDavidsonPilon** [comment]
`freq_multiplier` is better, as it is now aligned with the `freq` keyword in the other functions. 

PR lgtm, go ahread and rebase, and merge on 🍏!

**Comment by aprotopopov** [comment]
- squashed some commits. Now history should be clearer. 
- also tried to fix `pydocstyle` and `pycodestyle` issues (some of `flake8` warnings about variables are still exist).

**Comment by CamDavidsonPilon** [comment]
bah more mpl failures. I think let's ignore them, and I'll make an issue to find a better long-term solution for the mpl tests.

**Comment by CamDavidsonPilon** [comment]
PR lgtm, thanks @aprotopopov 

I think we should start to prep a new sub-version release as well.

**Pull request merged**